src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageFeed.cs

```diff
@@ -174,6 +242,24 @@ public async Task<SearchResult<IPackageSearchMetadata>> RefreshSearchAsync(Refre
                 }
             }
 
+            if (_telemetryService != null
```

Comment:
can't we just simplified it to raise telemetry event (if required) at the end of this method for each page?

I wasn't able to understand a. why `TelemetryState` exists? and b. why it's part of `AggregateSearchResultsAsync`?

Reply:
Spoke offline. `TelemetryState` is to carry operation ID, index, and duration telemetry data across multiple pages and when `RefreshSearchAsync` is called multiple times.